### PR TITLE
Generate only kpis

### PR DESCRIPTION
**Pull request by marota**
You can now generate KPIs based on existing data as with this cli
 ```
 chronix2grid --mode K --output-folder pathTo/data_grid2op/l2rpn_wcci_2022/chronics --input-folder ChroniX2Grid_path/input_data --ignore-warnings --weeks 4 --case case118_l2rpn_wcci_2022 --n_scenarios 1 
```
You can also only generate KPIs for your generated data if you don't want to bother generating KPIs for some reference data. This can be done by setting in ParamsKPI.json
 ```
"comparison":null 
 ```

### CODE REVIEW

File: input_data/generation/case118_l2rpn_wcci_2022/remove_load.py
Comment (BDonnot) [inline input_data/generation/case118_l2rpn_wcci_2022/remove_load.py:24]: probably not great to have this file here either

File: requirements.txt
Comment (BDonnot) [inline requirements.txt:13]: completely outdated (I modified the code in the branch `ramp_forecast` to use up to date versions of these libraries, I can make the PR and we merge this after if you prefer.

File: tests/data/input/generation/case118_l2rpn_wcci_2022/gen_info.txt
Comment (BDonnot) [inline tests/data/input/generation/case118_l2rpn_wcci_2022/gen_info.txt:11]: this file has nothing to do here either

**Review by BDonnot — CHANGES_REQUESTED** [review]
I think some changes need to be made before merging, most notably:

- the plt.ioff() or something like that, which can have side effects impossible to find and debug
- the requirements.txt that uses completely outdated library (in the branch `ramp_forecasts` i made the changes to use most recent versions and tests pass, maybe we could merge this branch first ?)
- some minor readability issues with the `if not`

File: README.md
Comment (marota) [reply README.md:193]: Ok done

File: README.md
Comment (marota) [reply README.md:91]: I created an issue for improvement to be able to directly do this from grid2op indeed

File: chronix2grid/config.py
Comment (marota) [reply chronix2grid/config.py:177]: Good point, it was already there before. As I am not sure which exception to use here and it is not per say related to this PR, I create this issue to keep track of it https://github.com/BDonnot/ChroniX2Grid/issues/81

File: chronix2grid/kpi/deterministic/kpis.py
Comment (marota) [reply chronix2grid/kpi/deterministic/kpis.py:24]: Ok not sure why it was there, I commented it

File: chronix2grid/kpi/deterministic/kpis.py
Comment (marota) [reply chronix2grid/kpi/deterministic/kpis.py:96]: The most general case comes first as you would expect normally some reference data for comparison. I would keep it like that for this reason, even if I agree with you in genera

File: chronix2grid/main.py
Comment (marota) [reply chronix2grid/main.py:235]: Good point, I added an issue https://github.com/BDonnot/ChroniX2Grid/issues/82

File: requirements.txt
Comment (marota) [reply requirements.txt:13]: removed

File: chronix2grid/kpi/deterministic/kpis.py
Comment (marota) [reply chronix2grid/kpi/deterministic/kpis.py:161]: I commented it as not sure what it is used for

File: chronix2grid/kpi/main.py
Comment (marota) [reply chronix2grid/kpi/main.py:124]: Ok removed it

File: chronix2grid/kpi/main.py
Comment (marota) [reply chronix2grid/kpi/main.py:135]: Ok removed it

File: chronix2grid/kpi/main.py
Comment (marota) [reply chronix2grid/kpi/main.py]: Definitely beeter thanks and done

File: chronix2grid/kpi/preprocessing/pivot_KPI.py
Comment (marota) [reply chronix2grid/kpi/preprocessing/pivot_KPI.py:71]: ok merci, done

File: chronix2grid/kpi/preprocessing/pivot_KPI.py
Comment (marota) [reply chronix2grid/kpi/preprocessing/pivot_KPI.py:146]: Yes indeed there was some unecessary code here, i changed it with your proposal

File: chronix2grid/kpi/preprocessing/pivot_utils.py
Comment (marota) [reply chronix2grid/kpi/preprocessing/pivot_utils.py:373]: Good point, done

File: chronix2grid/kpi/preprocessing/pivot_utils.py
Comment (marota) [reply chronix2grid/kpi/preprocessing/pivot_utils.py:363]: Indeed, done

File: chronix2grid/main.py
Comment (marota) [reply chronix2grid/main.py:114]: Yes it is really see, meaning if we are in a mode in which data is generated and not only computing kpis
I extended a bit the comment for better clarity there

File: chronix2grid/main.py
Comment (marota) [reply chronix2grid/main.py:143]: Ok thanks done !

File: tests/data/input/generation/case118_l2rpn_wcci_2022/gen_info.txt
Comment (marota) [reply tests/data/input/generation/case118_l2rpn_wcci_2022/gen_info.txt]: Removed

File: tests/data/input/generation/case118_l2rpn_wcci_2022/remove_load.py
Comment (marota) [reply tests/data/input/generation/case118_l2rpn_wcci_2022/remove_load.py]: Removed

File: input_data/generation/case118_l2rpn_wcci_2022/remove_load.py
Comment (marota) [reply input_data/generation/case118_l2rpn_wcci_2022/remove_load.py]: Removed

File: input_data/generation/case118_l2rpn_wcci_2022/gen_info.txt
Comment (marota) [reply input_data/generation/case118_l2rpn_wcci_2022/gen_info.txt]: Removed

**Comment by marota** [comment]
@BDonnot thanks for the long review, I addressed all comments and merged master into it to be up to date.
I think this PR can be merged, so you can have a last quick check

**Pull request merged**
